Export NavigationGuardProviderContext

jackstenglein opened the pull request:
This allows users of this library to hook into the guardMapRef value and build their own custom functionality on top of it.

jackstenglein committed:
Export NavigationGuardProviderContext

peinguin commented:
I would like to see README changed in this PR to show how to use this new export

jackstenglein committed:
Update README.md

Adds instructions on using exported NavigationGuardProviderContext and clarifies custom dialog example

jackstenglein commented:
@peinguin Updated the README

peinguin reviewed (approved):
To be honest, I do not see why need to use **next**-navigation-guard if there is "a special routing setup that prevents you from using the built-in **Next**JS router".

However, I think, it will harm no one to get a list of active guards.
